src/mapping/grouping/pairwise_comparison.jl

```diff
@@ -0,0 +1,116 @@
+export GroupViaPairwiseComparison
+
+"""
+    GroupViaPairwiseComparison(; distance_threshold::Real, kwargs...)
+
+Initialize a struct that contains instructions on how to group features in
+[`AttractorsViaFeaturizing`](@ref). `GroupViaPairwiseComparison` groups features and
+identifies clusters by considering the pairwise distance between features. It can be used
+as an alternative to the clustering method in `GroupViaClustering`, having the
+advantage that it is simpler, typically faster and uses less memory. 
+
+## Keyword arguments
+    
+* `distance_threshold`: a real number defining the maximum distance two features can be to
+  be considered in the same cluster - above the threshold, features are different. This
+  value simply needs to be large enough to differentiate clusters.
+* `distance_metric = Euclidean()`: A metric to be used in the clustering. It can be any
+  function `f(a, b)` that returns the distance between any type of data structure (usually
+  vectors or matrices of reals). Needs to be consistent with the `featurizer` function. All
+  metrics from Distances.jl can be used here.
+* `rescale_features = true`: if true, rescale each dimension of the extracted features
```

Comment:
You argued that this should be false but here it is true (and you also say it leads to more accurate clustering)?

Reply:
I kept the same text as in the the clustering method haha. We can keep it like this, so the texts are consistent, and then discuss if the rescaling is appropriate or not separately. I also realized I exaggerated before, rescaling is actually fine. The biggest problem I had with rescaling doesn't happen now, because we are rescaling differently than before, when I had problems with it. There's only a minor case we need to deal with, I'll open a PR about this.